documentation/general/workloads/workload-history.md

```diff
@@ -34,11 +34,11 @@ The ID will be an identifier for the record which will start at 1 for the oldest
 
 ## `dotnet workload update --from-history
 
-A `--from-history` option will be added to the `dotnet workload update` command, which will update to a state from the workload history.  This will require one of two new parameters, `--before` or `--after`, to be specified.  By default the command will update the manifests to the versions specified, as well as install or uninstall workloads to match the workloads that were installed at the specified point.  A `--manifests-only` option will be supported which will update the manifests, but not change the current workloads that are installed (this may result in installing or uninstalling packs to match the new manifest versions).
+A `--from-history` option will be added to the `dotnet workload update` command, which will update to the state after a workload history record was written. By default the command will update the manifests to the versions specified, as well as install or uninstall workloads to match the workloads that were installed at the specified point. A `--manifests-only` option will be supported which will update the manifests, but not change the current workloads that are installed (this may result in installing or uninstalling packs to match the new manifest versions).
```

Comment:
Thanks for updating this. To confirm then, the output of the history will look like what's above in this .md file? Are there any sample outputs in the discussion that I could look at? Additionally, sounds like now we limit the behavior to only after a specific update is listed. I assume that's because we're really only recording the action and end state we got to after a command was run, correct?

Reply:
> Thanks for updating this. To confirm then, the output of the history will look like what's above in this .md file? Are there any sample outputs in the discussion that I could look at?

Yes; it should look like that. I don't think I attached any screenshots previously, but here's what a sample output I just made:
```
>dotnet workload history

Id      Date                              Command      Workloads
----------------------------------------------------------------
1       1/29/2024 4:15:20 PM -08:00       update
2       1/30/2024 12:42:51 PM -08:00      update
3       2/1/2024 1:57:49 PM -08:00        update
```

After I took that, I called `dotnet workload install maui-android`, and it changed to this:
```
>dotnet workload history

Id      Date                              Command      Workloads
-------------------------------------------------------------------
1       1/29/2024 4:15:20 PM -08:00       update
2       1/30/2024 12:42:51 PM -08:00      update
3       2/1/2024 1:57:49 PM -08:00        update
4       2/1/2024 1:59:38 PM -08:00        install      maui-android
```

Additionally, sounds like now we limit the behavior to only after a specific update is listed. I assume that's because we're really only recording the action and end state we got to after a command was run, correct?

Not exactly. We still record both what the state was before and what the state was after—which is important if we want to be able to insert 'unknown' markers. The change here was for clarity on the CLI.

Reply:
This format looks great, I really like this and I think it will be rather useful. :)
That said, I think it'd also be good to contain a bit more information from the history record on here, namely what other arguments were used. For example if they did workload clean, did they use the --all flag? Was this decided against being valuable enough to put here?

Reply:
I think the argument (hehe) against including the full command line is that it can get long in some cases, and it would look awkward if it gets split onto multiple lines. The nice thing about Id/Date/Command/Workloads is that they're all (well, mostly) guaranteed to be pretty short.

I do think it could be useful to have some extra arguments, though. Perhaps it could display full command line except the first three words (i.e., 'dotnet workload update') but also cut it off if it's longer than...20? characters and replace the last three with '...' What do you think of that design? (And I obviously just threw out 20, so if you think that number should be different, feel free to opine.)

Reply:
@Forgind I think logging the whole command line in the history is a good move, but for display purposes I think if you're using the 'Table' abstraction we have it should auto-truncate the display.  One thing I was also thinking about was we may want a `--format <json|table>` argument the way that tool list and a few other commands have, for tooling purposes. That would make it easier to justify truncating the display in the Table format.